main.cpp

```diff
@@ -126,172 +120,108 @@ i cubed: 531441
+    template<typename OtherType>
+    Numeric& operator*=( const OtherType& rhs )
     {
-        *a *= rhs;
+        *a *= static_cast<T>(rhs);
         return *this;
     }
 
-    Numeric& operator/=( Primitive rhs )
+    template<typename OtherType>
+    Numeric& operator/=( const OtherType& rhs )
     {   
         if constexpr ( std::is_same<Primitive, int>::value )
         {
-            if constexpr ( std::is_same< decltype(rhs), int>::value )
+            if constexpr ( std::is_same< OtherType, int>::value )
             { 
                 if ( rhs == 0 )
                 {
                     std::cout << "Can't divide by 0.\n";
                     return *this;
                 }
-                else if ( rhs < std::numeric_limits<Primitive>::epsilon() )
+                else if ( rhs < std::numeric_limits<OtherType>::epsilon() )
                 {
                     std::cout << "Can't divide by 0.\n";
                     return *this;
                 }
             }
         }
-        else if ( rhs < std::numeric_limits<Primitive>::epsilon() )
+        else if ( rhs < std::numeric_limits<OtherType>::epsilon() )
         {
             std::cout << "Warning, dividing by 0.\n";
         }
 
-        *a /= rhs;
+        *a /= static_cast<T>(rhs);
         return *this;
     } 
 
 private:
     std::unique_ptr<Primitive> a;
-
-    Numeric& powInternal( const Primitive value )
-    {
-        if( a != nullptr )
-            *a = static_cast<Primitive>( std::pow( *a, value ) ); // to avoid warning when converting into <int> type
-        return *this;
-    }
-};
-
-// DoubleType Templated Def
-template <>
-struct Numeric<double>
-{
-    using Primitive = double;
-
-    Numeric( Primitive varA ) : a( new Primitive(varA) ) {}
-
-    operator Primitive() const { return *a; }
-
-    template<typename FuncPtr>
-    Numeric& apply( FuncPtr func )
-    {
-        func( *a );
-        return *this;
-    }
-
-    Numeric& pow( Primitive rhs )
-    {
-        return powInternal ( rhs );
-    }
-
-    Numeric& operator+=( Primitive rhs )
-    {
-        *a += rhs;
-        return *this;
-    }
-
-    Numeric& operator -=( Primitive rhs )
-    {
-        *a -= rhs;
-        return *this;
-    }
-
-    Numeric& operator *=( Primitive rhs )
-    {
-        *a *= rhs;
-        return *this;
-    }
-
-    Numeric& operator/=( Primitive rhs )
-    {
-        *a /= rhs;
-        return*this;
-    } 
-
-    private:
-    std::unique_ptr<Primitive> a;
-
-    Numeric& powInternal( const Primitive value )
-    {
-        if( a != nullptr )
-            *a = std::pow( *a, value );
-        return *this;
-    }
 };
 
 struct Point
 {
     Point( float a, float b ) : x( a ), y( b ) {}
 
-    Point& multiply(float m)
+    template<typename OtherType>
+    Point& multiply( const OtherType& m)
     {
-        x *= m;
-        y *= m;
+        x *= static_cast<float>(m);
+        y *= static_cast<float>(m);
         return *this;
     }
```

Comment:
I'm not sure that you needed to template this function.   Just cast whatever you're passing to it.   your `operator*=` does that cast as I would expect it to be seen in main()
- [ ] revise.